pkg/stagemanager/stage_manager.go

```diff
@@ -667,3 +652,62 @@ func (stm *StageManager) StopMosnProcess() (err error) {
 		return
 	}
 }
+
+// ReloadMosnProcess stops Mosn process via command line,
+// and it not support reload on windows.
+func (stm *StageManager) ReloadMosnProcess() (err error) {
+	// init
+	stm.runParamsParsedStage()
+	stm.runReconfigurationParamsInit()
+
+	mosnConfig := stm.data.config
+
+	log.StartLogger.Infof("[mosn reload] start reload mosn configurations")
+	// finds process and sends SIGINT to mosn process, makes it force exit.
+	proc, p, err := getMosnProcess(mosnConfig)
+	if err != nil {
+		log.StartLogger.Errorf("[mosn reload] fail to find process(%v), err: %v", p, err)
+		return
+	}
+
+	if err = sendSignal2Mosn(proc, p, syscall.SIGHUP); err != nil {
+		log.StartLogger.Errorf("[mosn reload] fail to send HUP signal to mosn process(%v), err: %v", p, err)
+		time.Sleep(100 * time.Millisecond) // waiting logs output
+		return
+	}
+
```

Comment:
we'd better wait the old process is quitted, and the new process is started, before return.

Reply:
ok, using signal to check process state or admin/state api?

Reply:
checking process state should be fine.

Reply:
Mosn process is always alive after I execute `mosn reload`
We seem couldn't check mosn process state by sending signal?